presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -158,6 +158,27 @@ public void testRoundTrip()
     }
 
     @Test
+    public void testInOptimization()
+            throws Exception
+    {
+        TupleDomain<Symbol> tupleDomain = withColumnDomains(ImmutableMap.<Symbol, Domain>builder()
+                .put(A, Domain.create(ValueSet.ofRanges(Range.lessThan(BIGINT, 1L), Range.range(BIGINT, 1L, false, 2L, false), Range.range(BIGINT, 2L, false, 3L, false), Range.greaterThan(BIGINT, 3L)), false))
+                .build());
+
+        assertEquals(toPredicate(tupleDomain), not(in(A, ImmutableList.of(1L, 2L, 3L))));
+
+        Domain testDomain = Domain.create(ValueSet.ofRanges(
+                Range.lessThan(BIGINT, 1L),
+                Range.range(BIGINT, 1L, false, 2L, false),
+                Range.range(BIGINT, 2L, false, 3L, false),
```

Comment:
you might be able to represent this domain construction in a more readable fashion by constructing it from the combination of union/intersect/complement etc.